### PR TITLE
Fix sending timestamp in client streaming

### PR DESCRIPTION
**Pull request by dilanSachi**
## Purpose
$title
## Examples
N/A
## Checklist
- [x] Linked to an issue
- [ ] Updated the changelog
- [x] Added tests

Fixes https://github.com/ballerina-platform/ballerina-standard-library/issues/1733

### CODE REVIEW

File: ballerina/tests/47_unary_timestamp_client.bal [hunk 1]
```diff
@@ -18,37 +18,102 @@ import ballerina/test;
...
+@test:Config {enable: true}
+function testTimestampServerStream() returns error? {
+    TimestampServiceClient utsClient = check new ("http://localhost:9147");
+    time:Utc sendingTime = check time:utcFromString("2008-12-03T11:15:30.120Z");
+
+    var result = check utsClient->serverStreamTime(sendingTime);
+    int count = 0;
+    time:Utc[] receivedData = [];
+    error? e = result.forEach(function(time:Utc value) {
+        receivedData[count] = <time:Utc>value;
```
Comment (BuddhiWathsala) [inline ballerina/tests/47_unary_timestamp_client.bal:68]: Do we need to cast the `value` here?

File: ballerina/tests/47_unary_timestamp_client.bal [hunk 2]
```diff
@@ -18,37 +18,102 @@ import ballerina/test;
...
+    time:Utc sendingTime = check time:utcFromString("2008-12-03T11:15:30.120Z");
+
+    ContextTimestampStream result = check utsClient->serverStreamTimeContext({
+        headers: {"req-header": ["1234567890", "2233445677"]},
+        content: sendingTime
+    });
+    int count = 0;
+    time:Utc[] receivedData = [];
+    error? e = result.content.forEach(function(time:Utc value) {
+        receivedData[count] = <time:Utc>value;
```
Comment (BuddhiWathsala) [inline ballerina/tests/47_unary_timestamp_client.bal:86]: Same, I think we don't need to cast.

File: ballerina/tests/47_unary_timestamp_service.bal [hunk 3]
```diff
@@ -50,4 +50,26 @@ service "UnaryTimestampService" on ep47 {
             check caller->complete();
         }
     }
+
+    remote function serverStreamTime(TimestampServiceTimestampCaller caller, time:Utc value) returns error? {
+        time:Utc responseTime = checkpanic time:utcFromString("2008-12-03T11:15:30.120Z");
```
Comment (BuddhiWathsala) [inline ballerina/tests/47_unary_timestamp_service.bal:55]: Let's replace `checkpanic` with `check`.

File: ballerina/tests/47_unary_timestamp_service.bal [hunk 4]
```diff
@@ -50,4 +50,26 @@ service "UnaryTimestampService" on ep47 {
...
+                headers: {},
+                content: val
+            });
+        });
+    }
+
+    remote function clientStreamTime(TimestampServiceTimestampCaller caller, stream<time:Utc, Error?> clientStream) returns error? {
+        time:Utc[] timearr = [];
+        error? e = clientStream.forEach(function(time:Utc value) {
+            timearr.push(checkpanic time:utcFromString("2008-12-03T11:15:30.120Z"));
```
Comment (BuddhiWathsala) [inline ballerina/tests/47_unary_timestamp_service.bal:68]: Same, let's replace the `checkpanic`.

File: ballerina/tests/47_unary_timestamp_pb.bal [hunk 5]
```diff
@@ -105,13 +105,116 @@ public isolated client class UnaryTimestampServiceClient {
         } else {
             message = req;
         }
-        var payload = check self.grpcClient->executeSimpleRPC("UnaryTimestampService/getBiTime", message, headers);
+        var payload = check self.grpcClient->executeSimpleRPC("TimestampService/exchangeTime", message, headers);
         [anydata, map<string|string[]>] [result, respHeaders] = payload;
         return {content: <time:Utc>result.cloneReadOnly(), headers: respHeaders};
     }
+
+    isolated remote function clientStreamTime() returns (ClientStreamTimeStreamingClient|Error) {
```
Comment (DimuthuMadushan) [inline ballerina/tests/47_unary_timestamp_pb.bal:113]: Do we need parentheses here in return type?

Comment (dilanSachi) [reply ballerina/tests/47_unary_timestamp_pb.bal:113]: This is generated from the protobuf tool
@BuddhiWathsala should we change the tool to not generate those parentheses?

Comment (BuddhiWathsala) [reply ballerina/tests/47_unary_timestamp_pb.bal:113]: Ideally, we should. @MadhukaHarith92 @daneshk  WDYT?.

File: ballerina/tests/47_unary_timestamp_service.bal (hunk 4 above)
Comment (dilanSachi) [reply ballerina/tests/47_unary_timestamp_service.bal]: Fixed in https://github.com/ballerina-platform/module-ballerina-grpc/pull/372/commits/69f160eca3668e350a01ca297ba3404aae9ce7d2

File: ballerina/tests/47_unary_timestamp_service.bal (hunk 3 above)
Comment (dilanSachi) [reply ballerina/tests/47_unary_timestamp_service.bal]: Fixed in https://github.com/ballerina-platform/module-ballerina-grpc/pull/372/commits/69f160eca3668e350a01ca297ba3404aae9ce7d2

File: ballerina/tests/47_unary_timestamp_client.bal (hunk 2 above)
Comment (dilanSachi) [reply ballerina/tests/47_unary_timestamp_client.bal]: Fixed in https://github.com/ballerina-platform/module-ballerina-grpc/pull/372/commits/69f160eca3668e350a01ca297ba3404aae9ce7d2

File: ballerina/tests/47_unary_timestamp_client.bal (hunk 1 above)
Comment (dilanSachi) [reply ballerina/tests/47_unary_timestamp_client.bal]: Fixed in https://github.com/ballerina-platform/module-ballerina-grpc/pull/372/commits/69f160eca3668e350a01ca297ba3404aae9ce7d2

File: ballerina/tests/47_unary_timestamp_service.bal (hunk 4 above)
Comment (dilanSachi) [reply ballerina/tests/47_unary_timestamp_service.bal]: Fixed the failing issue in https://github.com/ballerina-platform/module-ballerina-grpc/pull/372/commits/3dcc4e846f9601c42174bdf5d63a9766ee306da5

File: ballerina/tests/47_unary_timestamp_pb.bal (hunk 5 above)
Comment (daneshk) [reply ballerina/tests/47_unary_timestamp_pb.bal:113]: yes, @MadhukaHarith92 is parentheses coming along with syntax tree node or we are adding?

Comment (dilanSachi) [reply ballerina/tests/47_unary_timestamp_pb.bal:113]: I think we are adding them manually
https://github.com/ballerina-platform/module-ballerina-grpc/blob/3b433d0ccd4c4b08eecbd77018c8c26e33aacb27/native/src/main/java/io/ballerina/stdlib/grpc/builder/syntaxtree/utils/UnaryUtils.java#L165

**Review by daneshk — APPROVED** [review]

File: ballerina/tests/47_unary_timestamp_pb.bal (hunk 5 above)
Comment (BuddhiWathsala) [reply ballerina/tests/47_unary_timestamp_pb.bal:113]: Let's complete this as a separate task. https://github.com/ballerina-platform/ballerina-standard-library/issues/1739

**Review by BuddhiWathsala — APPROVED** [review]

File: ballerina/tests/47_unary_timestamp_pb.bal (hunk 5 above)
Comment (MadhukaHarith92) [reply ballerina/tests/47_unary_timestamp_pb.bal:113]: @daneshk, @BuddhiWathsala , yes we have added this parenthesis. I think that was the old Ballerina practice. +1 to remove those parentheses.

**Pull request merged**